docs/go-quickstart.md

```diff
@@ -0,0 +1,167 @@
+
+```go
+// This file is autogenerated. DO NOT EDIT.
+//
+// Generated by "thema lineage gen" from lineage defined in ship.cue
+
+package ship
+
+// Ship defines model for ship.
+type Ship struct {
+	FirstField  string `json:"firstField"`
+	SecondField int    `json:"secondField"`
+}
+```
+
+
+Cue types are more expressive as compared to Go types, To use the rich information from CUE in GO programs, Users can use Bindings. Using Bindings users get access to the thema Lineage defined in `ship.cue` and can start validating their data against any of the schema.
+
+## Generating Go Bindings
+
+Use the below command to generate a Go binding from `ship.cue`
+
+```
+thema lineage gen gobindings -l ship.cue > go_bindings.go
+```
+
+The genearted Go binding will look like below
+
+```go
+// This file is autogenerated. DO NOT EDIT.
+//
+// Generated by "thema lineage gen" from lineage defined in ship.cue
+
+package ship
+
+import (
+	"embed"
+	"path"
+
+	"github.com/grafana/thema"
+	"github.com/grafana/thema/load"
+)
+
+//go:embed ship.cue
+var themaFS embed.FS
+
+// Lineage returns a [thema.Lineage] for the 'ship' lineage.
+//
+// The lineage is the canonical specification of ship. It contains
+// all versions of the Thema schema that have ever existed for ship,
+// and the lenses that allow valid instances of one schema in the lineage to
+// be translated to another version of schema in the lineage.
+//
+// This function will return an error if the [Thema invariants] are not met by
+// the lineage defined in ship.cue.
+//
+// [Thema's general invariants]: https://github.com/grafana/thema/blob/main/docs/invariants.md
+func Lineage(rt *thema.Runtime, opts ...thema.BindOption) (thema.Lineage, error) {
+	// Load a build instance from the embedded fs
+	inst, err := load.InstancesWithThema(themaFS, path.Dir("ship.cue"))
+	if err != nil {
+		// Errors at this point indicate a problem with basic loading of .cue file bytes,
+		// which typically means the code generator was misconfigured and a path input
+		// is incorrect.
+		return nil, err
+	}
+
+	raw := rt.Context().BuildInstance(inst)
+
+	// Errors here indicate that:
+	//   - The parsed path does not exist in the loaded CUE file (["github.com/grafana/thema/errors".ErrValueNotExist])
+	//   - The value at the parsed path exists, but does not appear to be a Thema
+	//     lineage (["github.com/grafana/thema/errors".ErrValueNotALineage])
+	//   - The value at the parsed path exists and is a lineage (["github.com/grafana/thema/errors".ErrInvalidLineage]),
+	//     but is invalid due to the violation of some general Thema invariant -
+	//     for example, declared schemas don't follow backwards compatibility rules,
+	//     lenses are incomplete.
+	return thema.BindLineage(raw, rt)
+}
+
+// type guards
+
+var _ thema.LineageFactory = Lineage
+```
+This sets us up where we have a single Go function, Which we can call and it will load up the `ship.cue` from disk and return a thema Lineage
```

Comment:
```suggestion
We now have a single Go function our program can call, and it will load up the `ship.cue` from disk and return a [`thema.Lineage`](https://pkg.go.dev/github.com/grafana/thema#Lineage)
```